Detectors/TRD/simulation/include/TRDSimulation/TRsim.h

```diff
@@ -0,0 +1,134 @@
+
+/* $Id$ */
+
+////////////////////////////////////////////////////////////////////////////
+//                                                                        //
+//  TRD simulation - multimodule (regular rad.)                           //
+//                                                                        //
+////////////////////////////////////////////////////////////////////////////
+#ifndef ALICEO2_TRD_TRSIM_H_
+#define ALICEO2_TRD_TRSIM_H_
+
+#include <TObject.h>
+#include <TMath.h>
+
+class TH1D;
+
+class TRsim : public TObject
+{
+ public:
+  TRsim();
+  ~TRsim();
+  void init();
+  int createPhotons(int pdg, float p, std::vector<float>& ePhoton);
+  int calculatePhotons(float p, float mass, std::vector<float>& ePhoton);
+  double getSigma(double energykeV);
+  double interpolate(double energyMeV, double* en, const double* const mu, int n);
+  int locate(double* xv, int n, double xval, int& kl, double& dx);
+  double getOmega(float rho, float z, float a) { return (28.8 * TMath::Sqrt(rho * z / a)); };
+  int selectNFoils(float p) const;
+
+  void setFoilThick(float t)
+  {
+    mFoilThick = t;
+    setSigma();
+  };
+  void setGapThick(float t)
+  {
+    mGapThick = t;
+    setSigma();
+  };
+  void setFoilDens(float d)
+  {
+    mFoilDens = d;
+    mFoilOmega = getOmega(mFoilDens, mFoilZ, mFoilA);
+    setSigma();
+  };
+  void setFoilZ(float z)
+  {
+    mFoilZ = z;
+    mFoilOmega = getOmega(mFoilDens, mFoilZ, mFoilA);
+  };
+  void setFoilA(float a)
+  {
+    mFoilA = a;
+    mFoilOmega = getOmega(mFoilDens, mFoilZ, mFoilA);
+  };
+  void setGapDens(float d)
+  {
+    mGapDens = d;
+    mGapOmega = getOmega(mGapDens, mGapZ, mGapA);
+    setSigma();
+  };
+  void setGapZ(float z)
+  {
+    mGapZ = z;
+    mGapOmega = getOmega(mGapDens, mGapZ, mGapA);
+  };
+  void setGapA(float a)
+  {
+    mGapA = a;
+    mGapOmega = getOmega(mGapDens, mGapZ, mGapA);
+  };
+  void setTemp(float t)
+  {
+    mTemp = t;
+    setSigma();
+  };
+  void setSigma();
+
+  double getMuPo(double energyMeV);
+  double getMuCO(double energyMeV);
+  double getMuXe(double energyMeV);
+  double getMuAr(double energyMeV);
+  double getMuMy(double energyMeV);
+  double getMuN2(double energyMeV);
+  double getMuO2(double energyMeV);
+  double getMuHe(double energyMeV);
+  double getMuAi(double energyMeV);
+
+  float getFoilThick() const { return mFoilThick; };
+  float getGapThick() const { return mGapThick; };
+  float getFoilDens() const { return mFoilDens; };
+  float getGapDens() const { return mGapDens; };
+  double getFoilgetOmega() const { return mFoilOmega; };
+  double getGapgetOmega() const { return mGapOmega; };
+  float getTemp() const { return mTemp / 273.16; };
+  TH1D* getSpectrum() const { return mSpectrum; };
+
+ protected:
+  int mNFoilsDim;    //  Dimension of the NFoils array
+  int* mNFoils;      //[mNFoilsDim] Number of foils in the radiator stack
+  double* mNFoilsUp; //[mNFoilsDim] Upper momenta for a given number of foils
```

Comment:
could you modernize this to be std::array ?

Reply:
Done.